sdks/java/core/src/main/java/org/apache/beam/sdk/options/PipelineOptionsReflector.java

```diff
@@ -36,7 +36,8 @@ private PipelineOptionsReflector() {}
    *
    * @see PipelineOptionsReflector#getOptionSpecs(Iterable)
    */
-  static Set<PipelineOptionSpec> getOptionSpecs(Class<? extends PipelineOptions> optionsInterface) {
+  static Set<PipelineOptionSpec> getOptionSpecs(
+      Class<? extends PipelineOptions> optionsInterface, boolean skipHidden) {
```

Comment:
This introduces a new feature: hidden options. I'm not sure about the consequences of such a feature. I believe such options would be better stored in the docker environment config.